libs/core/src/main/java/org/opensearch/core/xcontent/filtering/FilterPath.java

```diff
@@ -99,32 +101,31 @@ public static FilterPath[] compile(Set<String> filters) {
 
         List<FilterPath> paths = new ArrayList<>();
         for (String filter : filters) {
-            if (filter != null) {
+            if (filter != null && !filter.isEmpty()) {
                 filter = filter.trim();
                 if (filter.length() > 0) {
-                    paths.add(parse(filter, filter));
+                    paths.add(parse(filter));
+                } else {
+                    logger.warn("Filter is empty!");
```

Comment:
Is this helpful? What's one to do with this warning? Should it point to which filter is problematic?

Reply:
Not really helpful to TBH. Will remove